source/docs/software/wpilib-tools/robot-simulation/unit-testing.rst

```diff
@@ -0,0 +1,91 @@
+Unit Testing
+============
+
+Unit testing is a method of testing code by dividing the code into the smallest "units" possible and testing each unit. In robot code, this can mean testing the code for each subsystem individually.
+There are many unit testing frameworks for most languages. Java robot projects have `JUnit 4<https://junit.org/junit4/>`__ available by default, and C++ robot projects have `Google Test<https://github.com/google/googletest/blob/master/docs/primer.md>`__.
+
+Let's write our subsystem code, and then we'll write a test for it.
```

Comment:
Can we have a non-commandbased example, otherwise this should be in the commandbased section

Reply:
It isn't command-based, maybe the terminology is confusing. I want to specify that while the base example isn't command-based, it is very easy to use in command-based. I plan to specify what exactly would need to be changed (one instance of this is already in the doc).

Reply:
Yeah, that sounds good.